src/msg/async/AsyncConnection.cc

```diff
@@ -967,17 +967,31 @@ int AsyncConnection::_process_connection()
           ::close(sd);
         }
 
-        sd = net.connect(get_peer_addr());
+        sd = net.nonblock_connect(get_peer_addr());
         if (sd < 0) {
           goto fail;
         }
-        r = net.set_nonblock(sd);
+        center->create_file_event(sd, EVENT_READABLE, read_handler);
+        state = STATE_CONNECTING_RE;
+        break;
+      }
+
+    case STATE_CONNECTING_RE:
+      {
+        if (reconnect == false) {
```

Comment:
Yes, you are right. So, we should remove "bool reconnect", and make loop stay at "STATE_CONNECTING_RE" when "state == STATE_CONNECTING_RE" in fault()?